workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Utility/EntityBuilderTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using NUnit.Framework;
+using Improbable.Worker.Core;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class EntityBuilderTest
+    {
+        [Test]
+        public void EntityBuilder_should_throw_exception_if_no_position_component_is_added()
+        {
+            var builder = EntityBuilder.Begin();
+            Assert.Throws<InvalidEntityException>(() => builder.Build());
+        }
+
+        [Test]
+        public void EntityBuilder_should_throw_exception_if_component_is_added_twice()
+        {
+            var builder = EntityBuilder.Begin();
+            builder.AddPosition(0, 0, 0, "write-access");
+            Assert.Throws<InvalidOperationException>(() => builder.AddPosition(0, 0, 0, "write-access"));
```

Comment:
🤔 shall we have specific errors for this and the other `InvalidOperationException` instead?

Reply:
I'm not sure that would give the exception any more meaning - I think `InvalidOperationException` + the message give sufficient context.

Unless we expect users to want to catch and handle them differently? Although I can't see a use case off the top of my head